internal/providers/pluginfw/resources/sharing/resource_share.go

```diff
@@ -0,0 +1,396 @@
+	"github.com/databricks/databricks-sdk-go/service/sharing"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/sharing_tf"
+	"github.com/databricks/terraform-provider-databricks/logger"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+)
+
+const resourceName = "share"
+
+var _ resource.ResourceWithConfigure = &ShareResource{}
+
+func ResourceShare() resource.Resource {
+	return &ShareResource{}
+}
+
+type ShareInfoExtended struct {
+	sharing_tf.ShareInfo
+}
+
+func matchOrder[T any, K comparable](target, reference []T, keyFunc func(T) K) {
+	// Create a map to store the index positions of each key in the reference slice.
+	orderMap := make(map[K]int)
+	for index, item := range reference {
+		orderMap[keyFunc(item)] = index
+	}
+
+	// Sort the target slice based on the order defined in the orderMap.
+	sort.Slice(target, func(i, j int) bool {
+		return orderMap[keyFunc(target[i])] < orderMap[keyFunc(target[j])]
+	})
+}
+
+func suppressCDFEnabledDiff(si *sharing.ShareInfo) {
+	//suppress diff for CDF Enabled if HistoryDataSharingStatus is enabled , as API does not accept both fields to be set
+	for i := range si.Objects {
+		if si.Objects[i].HistoryDataSharingStatus == "ENABLED" {
+			si.Objects[i].CdfEnabled = false
+		}
+	}
+}
+
+func resourceShareMap(si sharing.ShareInfo) map[string]sharing.SharedDataObject {
+	m := make(map[string]sharing.SharedDataObject, len(si.Objects))
+	for _, sdo := range si.Objects {
+		m[sdo.Name] = sdo
+	}
+	return m
+}
+
+func equal(this sharing.SharedDataObject, other sharing.SharedDataObject) bool {
+	if other.SharedAs == "" {
+		other.SharedAs = this.SharedAs
+	}
+	//don't compare computed fields
+	other.AddedAt = this.AddedAt
+	other.AddedBy = this.AddedBy
+	other.Status = this.Status
+	other.HistoryDataSharingStatus = this.HistoryDataSharingStatus
+	other.ForceSendFields = this.ForceSendFields // TODO: is this the right thing to do?
+	return reflect.DeepEqual(this, other)
+}
+
+func diff(beforeSi sharing.ShareInfo, afterSi sharing.ShareInfo) []sharing.SharedDataObjectUpdate {
+	beforeMap := resourceShareMap(beforeSi)
+	afterMap := resourceShareMap(afterSi)
+	changes := []sharing.SharedDataObjectUpdate{}
+	// not in after so remove
+	for _, beforeSdo := range beforeSi.Objects {
+		_, exists := afterMap[beforeSdo.Name]
+		if exists {
```

Comment:
[nit] inline to make it clear that `exists` is not used outside of the condition. Also, it is more idiomatic to call such variable `ok`:
```suggestion
		if _, ok := afterMap[beforeSdo.Name]; ok {
```